### PR TITLE
handle home directory prefix

### PR DESCRIPTION
**Pull request by tharun634**
# Description

Introduced a util function that would take in the file path and resolve the `~/`  prefix, should it exist, to the absolute path of the user's home directory.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1245 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -350,6 +350,26 @@ func GetAbsPath(baseDir, path string) string {
 	return absPath
 }
 
+// ResolveHomeDir resolves prefix of the given path, if present, to the user's home directory and returns it
+func ResolveHomeDir(filePath string) (string, error) {
+	if filePath == "" {
+		return "", nil
+	}
+	homeDirPrefixes := []string{"~/", "$HOME/"}
```
Comment (pravinpushkar) [inline utils/utils.go:358]: AFAIK in windows this variable `$HOME` is called `Userprofile` and this logic might not be able to handle it. Should we avoid this part ?

**Comment by pravinpushkar** [comment]
@tharun634 Please fix the lint error to check the behaviour of your UTs on supported OSs.

File: utils/utils.go (hunk 1 above)
Comment (tharun634) [reply utils/utils.go:358]: Yes you are correct, I'll remove support for `$HOME` and support only `~/`

**Review by pravinpushkar — APPROVED** [review]

File: utils/utils.go [hunk 2]
```diff
@@ -350,6 +350,22 @@ func GetAbsPath(baseDir, path string) string {
 	return absPath
 }
 
+// ResolveHomeDir resolves prefix of the given path, if present, to the user's home directory and returns it.
+func ResolveHomeDir(filePath string) (string, error) {
+	if filePath == "" {
+		return "", nil
+	}
+	if homeDirPrefix := "~/"; strings.HasPrefix(filePath, homeDirPrefix) {
```
Comment (shubham1172) [inline utils/utils.go:358]: Can `homeDirPrefix := "~/"` be a constant?

Comment (mukundansundar) [reply utils/utils.go:358]: also is there a way to check this only for linux and posix like systems? As in Windows this will not be the case.

Comment (shubham1172) [reply utils/utils.go]: @mukundansundar on Windows also users can refer to %USERPROFILE%, do you want to not consider "~/" syntax? I propose to keep it the same as Linux and Mac.

image

**Comment by mukundansundar** [comment]
@tharun634 on an unrelated note, can you stop the automatic action run in your fork? I am getting notifications of the failures of those runs for example dapr-bot-schedule etc.....

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**